qrenderdoc/Windows/BufferViewer.cpp

```diff
@@ -6688,14 +6693,34 @@ void BufferViewer::on_highlightVerts_toggled(bool checked)
   INVOKE_MEMFN(RT_UpdateAndDisplay);
 }
 
+void BufferViewer::on_vtxExploderSlider_valueChanged(int value)
+{
+  m_Config.vtxExploderSliderSNorm = (float)value / 100.0f;
+
+  INVOKE_MEMFN(RT_UpdateAndDisplay);
+}
+
+void BufferViewer::on_vtxExploderSlider_sliderReleased()
+{
+  // Always snaps back to zero
+  ui->vtxExploderSlider->setSliderPosition(0);
```

Comment:
I went and added a slider reset button. Takes up a bit more space but I agree that it can help to be able to navigate around the mesh with it slightly exploded.

Reply:
I don't _dislike_ the auto-return behaviour necessarily and it can be useful sometimes, but I'm not sure the value is justified by the cost of keeping it implemented and having this atypical behaviour. As I think you've come around to as well it's not really feasible to provide both mechanisms. It ends up either being overcomplicated if there's an explicit UI control, or too hidden and confusing if there's some kind of shortcut like holding shift while dragging to make it temporary. Someone could do that by accident and get _really_ lost.

My intuition is that you really have to pick one way and stick to it, and although it's occasionally nice to have it return I don't see a sufficient benefit. I'd be fine if you wanted to maybe add a shortcut key for escape to reset while the slider is focussed - which it should be immediately after interacting with it. It's not discoverable and isn't quite fully automatic but it is at least something.